presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -125,19 +127,68 @@ public BaseJdbcClient(JdbcConnectorId connectorId, BaseJdbcConfig config, String
         if (config.getConnectionPassword() != null) {
             connectionProperties.setProperty("password", config.getConnectionPassword());
         }
-        this.mapTableNames = config.isMapLowercaseTableNames();
+
+        // generate the map of lowercased schema names to JDBC schema names
+        schemaMappingCache = CacheBuilder.newBuilder().build(new CacheLoader<String, Optional<String>>()
+        {
+            @Override
+            public Optional<String> load(String key)
+                    throws Exception
+            {
+                for (String schemaName : getOriginalSchemas()) {
+                    if (schemaName.equals(key)) {
+                        return Optional.of(schemaName);
+                    }
+
+                    String schemaNameLower = schemaName.toLowerCase(ENGLISH);
+                    if (schemaNameLower.equals(key)) {
```

Comment:
The key that is passed in is the schema name from Presto, which would already be lowercased. The first `IF` is (maybe) a shortcut in case the schema name is already lowercased. I can see just ignoring that and always lowercasing and then comparing. And at this point, this is loading from the DB, so that `getOriginalSchemas()` returns the schema names in their original from from the DB which might be all uppercased, or a mix of case, or possibly all lowercased depending on the DB implementation and configuration.

And if the schema name doesn't exist in the DB, then we can't have a NULL as the value as I discovered, thus using Optional to indicate a value or lack of value.